cli/azd/pkg/apphost/aca_ingress.go

```diff
@@ -0,0 +1,257 @@
+			bindingKey = fmt.Sprintf("%d", *binding.TargetPort)
+		}
+		endpointByContainerPort[bindingKey] = append(endpointByContainerPort[bindingKey], &bindingGroup{
+			Binding: binding,
+			name:    key.string,
+			order:   key.Index,
+		})
+	}
+	return endpointByContainerPort
+}
+
+// groupProperties iterate over the bindings from a group and returns the properties of the group.
+// Properties:
+// port, external, httpOnly, hasHttp2
+func groupProperties(endpointByTargetPort map[string][]*bindingGroup) map[string]*endpointGroupProperties {
+	endpointByTargetPortProperties := make(map[string]*endpointGroupProperties)
+	for containerPort, bindings := range endpointByTargetPort {
+		props := &endpointGroupProperties{httpOnly: true}
+		for _, binding := range bindings {
+			if binding.TargetPort != nil {
+				props.port = *binding.TargetPort
+			}
+			if binding.External {
+				props.external = true
+			}
+			if binding.Scheme != acaIngressSchemaHttp && binding.Scheme != acaIngressSchemaHttps {
+				props.httpOnly = false
+			}
+			if binding.Transport == acaIngressTransportHttp2 {
+				props.hasHttp2 = true
+			}
+		}
+		endpointByTargetPortProperties[containerPort] = props
+	}
+	return endpointByTargetPortProperties
+}
+
+// validateExternalBindings check the there is not more than one binding group exported or if there are non-http bindings
+// exported.
+func validateExternalBindings(endpointByTargetPortProperties map[string]*acaPort) error {
+	countExternalGroups := 0
+	for _, props := range endpointByTargetPortProperties {
+		if props.external {
+			countExternalGroups++
+		}
+		if props.external && !props.httpOnly {
+			return fmt.Errorf("External non-HTTP(s) endpoints are not supported")
+		}
+	}
+	if countExternalGroups > 1 {
+		return fmt.Errorf("Multiple external endpoints are not supported")
+	}
+	return nil
+}
+
+// endpointByTargetPortProperties try to find the right http ingress group to use.
+// If there is just one http group, it will be used.
+// If there are more than one http group, take the one which is external, and if there is no external one, take the group
+// which contains the first http binding referenced in the manifest (the one with the lowest index key).
+func httpIngress(endpointByTargetPortProperties map[string]*acaPort) string {
+	var httpOnlyGroups []string
+	for groupKey, props := range endpointByTargetPortProperties {
+		if props.httpOnly {
+			httpOnlyGroups = append(httpOnlyGroups, groupKey)
+		}
+	}
+
+	var ingress string
+	if len(httpOnlyGroups) == 1 {
+		ingress = httpOnlyGroups[0]
+	}
+
+	if ingress == "" {
+		// We have more than one, pick prefer external one
+		var externalHttpOnly []string
+		for _, httpGroup := range httpOnlyGroups {
+			if endpointByTargetPortProperties[httpGroup].external {
+				externalHttpOnly = append(externalHttpOnly, httpGroup)
+			}
+		}
+
+		if len(externalHttpOnly) == 1 {
+			ingress = externalHttpOnly[0]
+		} else if len(httpOnlyGroups) > 1 {
+			minIndex := 0
```

Comment:
The logic will have to change here if we adopt my idea where we have a slice of ordered keys - but I think it should be possible to do?